docker/driver.go

```diff
@@ -0,0 +1,96 @@
+package docker
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/bblfsh/performance"
+
+	errors "gopkg.in/src-d/go-errors.v1"
+	"gopkg.in/src-d/go-log.v1"
+)
+
+// One day it will become a config file but now it's to annoying to ship it
+var drivers = map[string]string{
+	"bash":       "bash-driver",
+	"cpp":        "cpp-driver",
+	"csharp":     "csharp-driver",
+	"go":         "go-driver",
+	"java":       "java-driver",
+	"javascript": "javascript-driver",
+	"php":        "php-driver",
+	"python":     "python-driver",
+	"ruby":       "ruby-driver",
+	"typescript": "typescript-driver",
+}
+
+// InstallDriver downloads driver's repository, checkouts to a given commit,
+// builds docker image and installs driver's image to bblfshd container.
+// Requires bblfshd container running.
+func InstallDriver(language, commit string) error {
+	if err := validateLanguage(language); err != nil {
+		return err
+	}
+
+	driver := drivers[language]
+	image := driverImage(driver, commit)
+	if err := downloadAndBuildDriver(driver, commit, image); err != nil {
+		return err
+	}
+	if err := installDriverToBblfshd(bblfshdContainer, language, image); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func downloadAndBuildDriver(driver, commit, image string) error {
+	tmpDir := filepath.Join(os.TempDir(), driver)
+	log.Debugf("creating directory %v", tmpDir)
+	if err := os.MkdirAll(tmpDir, os.ModePerm); err != nil {
+		return err
+	}
+
+	repo := driverUrl(driver)
+	log.Debugf("performing git clone repository %v %v", repo, commit)
+	cloneCommand := fmt.Sprintf("git clone %s %s ; cd %s ; git checkout %s", repo, tmpDir, tmpDir, commit)
+	if err := performance.ExecCmd(cloneCommand); err != nil {
+		return err
+	}
+
+	log.Debugf("building with image %v", image)
+	buildCommand := fmt.Sprintf("cd %s ; go run build.go %s", tmpDir, image)
+	if err := performance.ExecCmd(buildCommand); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Example of command: docker exec bblfshd-perf bblfshctl driver install python docker-daemon:bblfsh/python-driver:tag
+func installDriverToBblfshd(bblfshContainer, language, image string) error {
+	installCommand := fmt.Sprintf("docker exec %s bblfshctl driver install %s docker-daemon:%s",
+		bblfshContainer, language, image)
+
+	if err := performance.ExecCmd(installCommand); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func validateLanguage(language string) error {
+	if _, ok := drivers[language]; !ok {
+		return errors.NewKind("language %v is not supported").New(language)
+	}
+	return nil
+}
+
+func driverUrl(driver string) string {
+	return "https://github.com/bblfsh/" + driver + ".git"
```

Comment:
The `.git` suffix is fine, but not required.

Reply:
Done